### PR TITLE
Improving Notification Behavior

### PR DESCRIPTION
**Pull request by ruif3r**
# Description

Modified notification behavior by changing the service class for a worker class and an actual notification.

# ChangeType

Improving

# Steps to Test

Open the app

### CODE REVIEW

File: app/src/main/java/com/example/ncov19traking/BaseApp.kt [hunk 1]
```diff
@@ -31,10 +37,21 @@ class BaseApp : Application() {
...
+            ).apply { description = "Provide reminders and info" }
             val notificationManager = getSystemService(NotificationManager::class.java)
             notificationChannel?.let { notificationManager.createNotificationChannel(it) }
         }
+    }
+
+    private fun showNotificationByWorker() {
+        val workManager = WorkManager.getInstance(this)
+        val periodicWorkRequest =
+            PeriodicWorkRequestBuilder<NotificationWorker>(11, TimeUnit.HOURS).build()
```
Comment (jinmyjbv) [inline app/src/main/java/com/example/ncov19traking/BaseApp.kt]: this `11` could be in a constant.

**Review by jinmyjbv — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/example/ncov19traking/NotificationWorker.kt [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+        val pendingIntent = PendingIntent.getActivity(applicationContext, 0, intent, 0)
+        val builder = NotificationCompat.Builder(applicationContext, BaseApp.CHANNEL_ID)
+            .setSmallIcon(R.drawable.ic_notifications_black_24dp)
+            .setContentTitle("Reminder")
+            .setContentText("Remember to wash your hands for at least 20 seconds!")
+            .setStyle(
+                NotificationCompat.BigTextStyle()
+                    .bigText("Remember to wash your hands for at least 20 seconds!")
+            ).setPriority(NotificationCompat.PRIORITY_DEFAULT).setContentIntent(pendingIntent)
+            .setAutoCancel(true)
```
Comment (jinmyjbv) [inline app/src/main/java/com/example/ncov19traking/NotificationWorker.kt:45]: you should separate all the building of the `builder` in a separate method.

File: app/src/main/java/com/example/ncov19traking/NotificationWorker.kt [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+        with(NotificationManagerCompat.from(applicationContext)) {
+            notify(notificationId, builder.build())
+        }
+        return Result.success()
+    }
```
Comment (jinmyjbv) [inline app/src/main/java/com/example/ncov19traking/NotificationWorker.kt:46]: split this huge method into small readable methods.

File: build.gradle [hunk 4]
```diff
@@ -1,7 +1,7 @@
 // Top-level build file where you can add configuration options common to all sub-projects/modules.
 
 buildscript {
-    ext.kotlin_version = '1.3.70'
+    ext.kotlin_version = '1.3.72'
```
Comment (jinmyjbv) [inline build.gradle:4]: you could do the same you are doing for the kotlin version to all your gradle dependencies versions.

**Review by jinmyjbv — CHANGES_REQUESTED** [review]

**Comment by jinmyjbv** [comment]
For some reason the notification is not showing for me even if I change the code to 10 seconds for the worker

**Comment by ruif3r** [comment]
thats because the minimun amount of time to execute an operation that workManager allows is 15 minutes

**Review by jinmyjbv — APPROVED** [review]

**Pull request merged**